ci/courtesy-push/courtesy-push.js

```diff
@@ -26,114 +30,175 @@ function formatDeps(depArr) {
         const name = depDetails[1];
         const version = depDetails[3];
         console.log(name + ' ' + version);
-        newDepsDict[name] = version;
+
+        if (name.includes('_')) {
+            const basicName = name.split('_')[0];
+            if (!deps[basicName]) deps[basicName] = {};
+            if (!deps[basicName].configs) deps[basicName].configs = [];
+
+            deps[basicName].configs.push({ name, version, depStr: newDep });
+        } else {
+            if (!deps[name]) deps[name] = {};
+
+            deps[name].name = name;
+            deps[name].version = version;
+            deps[name].depStr = newDep;
+        }
     });
 
-    return newDepsDict;
+    return deps;
 }
 
-function findLastIndex(array, predicate) {
-    let l = array.length;
+/**
+ * The function removes all generated configs such as Node16/Node20 from the list of dependencies
+ * @param {Array} depsArray - array of parsed dependencies from UnifiedDependencies.xml
+ * @param {Object} depsForUpdate - dictionary of dependencies from getDeps method
+ * @param {Object} updatedDeps - structure to track added/removed dependencies
+ * @returns {Array} - updated array of dependencies and updatedDeps object { added: [], removed: []
+ */
+function removeConfigsForTasks(depsArray, depsForUpdate, updatedDeps) {
```

Comment:
Fixed the logic, now we are comparing the names of the updated packages to check if they have config or not, since the packages' names contain config inside based on the default task name